packages/ui/src/components/va-select/hooks/useMaxVisibleOptions.ts

```diff
@@ -0,0 +1,49 @@
+import { computed, ref, toRefs, watch, ExtractPropTypes } from 'vue'
+
+import type { SelectOption } from '../types'
+
+export const useMaxVisibleOptionsProps = {
+  maxVisibleOptions: { type: Number, default: 0 },
+}
+
+export const useMaxVisibleOptions = (
+  props: ExtractPropTypes<typeof useMaxVisibleOptionsProps> & { multiple: boolean, modelValue: SelectOption | SelectOption[] },
+) => {
+  const { modelValue } = toRefs(props)
```

Comment:
If here no plans to make another props reactive, then will be great to use `toRef()`.